src/acl/external/eDirectory_userip/ext_edirectory_userip_acl.cc

```diff
@@ -777,24 +783,76 @@ BindLDAP(edui_ldap_t *l, char *dn, char *pw, unsigned int t)
     }
 }
 
+// XXX: duplicate (partial) of Ip::Address::lookupHostIp
+/**
+ * Convert the IP address string representation in src to
+ * its binary representation.
+ *
+ * \return binary representation of the src IP address.
+ *         Must be free'd using freeaddrinfo().
+ */
+static struct addrinfo *
+makeIpBinary(const char *src)
+{
+    struct addrinfo want;
+    memset(&want, 0, sizeof(want));
+    want.ai_flags = AI_NUMERICHOST; // prevent actual DNS lookups!
+
+    struct addrinfo *dst = nullptr;
+    if (getaddrinfo(src, nullptr, &want, &dst) != 0) {
+        // not an IP address
+        /* free any memory getaddrinfo() dynamically allocated. */
+        if (dst) {
+            freeaddrinfo(dst);
+            dst = nullptr;
```

Comment:
We can simplify/clarify a little by removing this assignment, but I do not insist on this change because it (slightly) increases the risk of `dst` dereference in some future version of this code.

You can also simplify by returning `dst` immediately (on zero getaddrinfo) and then handling error cleanup. I do not insist on this change either.

Reply:
done.